beam/beam/handling_unit.py

```diff
@@ -42,9 +42,12 @@ def generate_handling_units(doc, method=None):
 			in ("Material Transfer", "Send to Subcontractor", "Material Transfer for Manufacture")
 			and row.handling_unit
 		):
-			handling_unit = frappe.new_doc("Handling Unit")
-			handling_unit.save()
-			row.to_handling_unit = handling_unit.name
+			if settings.always_generate_for_transfer:
+				handling_unit = frappe.new_doc("Handling Unit")
+				handling_unit.save()
+				row.to_handling_unit = handling_unit.name
+			else:

```

Comment:
This should include a condition where the stock quantities are equal. If you are splitting a HU into two pieces, they cannot have the same number.